### ISSUE TITLE
DNSCryptClient won't start

### ISSUE THREAD

**Issue by BooBerry** [F1ash/dnscrypt-proxy-gui#7]
Using Arch Linux with GNOME 64-bit, attempting to start the app results in this in the Terminal;

```
** (DNSCryptClient:11335): WARNING **: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-WIabpadbLy: Connection refused
QLayout: Attempting to add QLayout "" to InfoPanel "", which already has a layout
QLayout: Attempting to add QLayout "" to InfoPanel "", which already has a layout
QLayout: Attempting to add QLayout "" to MainWindow "", which already has a layout
```

Any ideas?

**Comment by F1ash** [comment]
it is a DBus problems.

**Comment by nsiregar** [comment]
Using Solus Budgie 64-bit, attempting to start via terminal resulted: 

```
QLayout: Attempting to add QLayout "" to InfoPanel "", which already has a layout
QLayout: Attempting to add QLayout "" to InfoPanel "", which already has a layout
QLayout: Attempting to add QLayout "" to MainWindow "", which already has a layout
```

without any further messages, the app not launched. App installed at 
```
/usr/bin/DNSCryptClient
/usr/lib/systemd/system/DNSCryptClient@.service
/usr/lib/systemd/system/DNSCryptClient_test@.service
/usr/lib64/kf5/kauth/dnscrypt_client_helper
/usr/lib64/kf5/kauth/dnscrypt_client_reload_helper
/usr/lib64/kf5/kauth/dnscrypt_client_test_helper
/usr/share/applications/DNSCryptClient.desktop
/usr/share/dbus-1/system-services/pro.russianfedora.dnscryptclient.service
/usr/share/dbus-1/system-services/pro.russianfedora.dnscryptclientreload.service
/usr/share/dbus-1/system-services/pro.russianfedora.dnscryptclienttest.service
/usr/share/dbus-1/system.d/pro.russianfedora.dnscryptclient.conf
/usr/share/dbus-1/system.d/pro.russianfedora.dnscryptclientreload.conf
/usr/share/dbus-1/system.d/pro.russianfedora.dnscryptclienttest.conf
/usr/share/icons/hicolor/64x64/apps/DNSCryptClient.png
/usr/share/knotifications5/DNSCryptClient.notifyrc
/usr/share/polkit-1/actions/pro.russianfedora.dnscryptclient.policy
/usr/share/polkit-1/actions/pro.russianfedora.dnscryptclientreload.policy
/usr/share/polkit-1/actions/pro.russianfedora.dnscryptclienttest.policy
```

**Comment by F1ash** [comment]
Tray icon is showed?

**Comment by nsiregar** [comment]
it is showed for a second then gone

**Comment by F1ash** [comment]
if segfault message not exist then app is works. Maybe need to correct Plasma System Tray Settings, because not all tray icons showed automatically.

**Comment by nsiregar** [comment]
How to do correct Plasma System Tray Settings?
Is it when compiling the source?

Trying to run in `gdb`, give messages that process exited normally.

**Comment by F1ash** [comment]
https://imgur.com/h2a5AMF 
https://imgur.com/hcUMxi7

**Comment by nsiregar** [comment]
Looks like we don't have those settings in budgie

**Comment by F1ash** [comment]
strangly... this is a standard KDE Plasma Settings...

So when you start app in Terminal then app showed and closed (terminal string come back) or app showed and hidden?

**Comment by omni6** [comment]
seems to happen only on GTK desktops, maybe this helps:
[1](https://forum.qt.io/topic/19152/qlayout-attempting-to-add-qlayout)
[2](https://forum.qt.io/topic/53190/qlayout-attempting-to-add-qlayout-to-mainwindow-which-already-has-a-layout-solved/4)

**Comment by nsiregar** [comment]
Sorry for late reply, the app doesn't show at all. Only show the systray icon for a second then gone.

**Comment by F1ash** [comment]
Hmmm... This problem not with layouts, because this is standard Qt warnings about incorrect widgets usage (but it is not critical). Need to read initiation code of app. Some later...

**Comment by omni6** [comment]
you think https://github.com/F1ash/dnscrypt-proxy-gui/commit/24604bd584a62eb4336095f0c6aab44388b06156 solved this issue?

**Comment by F1ash** [comment]
i think that last commit not solve the issue... Maybe some error at initiation of app. But in KDE all works..

**Comment by unixfox** [comment]
I've the same issue on Plasma 5.11.

**Comment by F1ash** [comment]
at first start?

**Comment by unixfox** [comment]
Yes

**Comment by F1ash** [comment]
can you to give terminal error messages at first start? (before this delete ~/$USER/.config/DNSCryptClient/DNSCryptClient.conf). I have not a problem at first start.

**Comment by unixfox** [comment]
```
QLayout: Attempting to add QLayout "" to InfoPanel "", which already has a layout
QLayout: Attempting to add QLayout "" to InfoPanel "", which already has a layout
QLayout: Attempting to add QLayout "" to MainWindow "", which already has a layout
```
And I forgot to add that I'm running the 1.11.10 version.

**Comment by baimafeima** [comment]
Any updates regarding the starting problem? @F1ash

**Comment by F1ash** [comment]
no...

**Comment by mxmilkiib** [comment]
Got the same, Arch Linux, i3 is my wm with the i3bar systray.
```
qt5ct: using qt5ct plugin
QLayout: Attempting to add QLayout "" to InfoPanel "", which already has a layout
QLayout: Attempting to add QLayout "" to InfoPanel "", which already has a layout
QLayout: Attempting to add QLayout "" to MainWindow "", which already has a layout
qt5ct: D-Bus system tray: no
```
Might it be a dependency, or do I actually have to be using KDE DE to run DNSCryptClient?

**Comment by F1ash** [comment]
maybe... all required libs (in Fedora, rawhide):
`# rpm -qR dnscrypt-proxy-gui`
/bin/sh
dnscrypt-proxy
hicolor-icon-theme
kf5-kauth
kf5-knotifications
libKF5Auth.so.5()(64bit)
libKF5CoreAddons.so.5()(64bit)
libKF5Notifications.so.5()(64bit)
libQt5Core.so.5()(64bit)
libQt5Core.so.5(Qt_5)(64bit)
libQt5Core.so.5(Qt_5.9)(64bit)
libQt5DBus.so.5()(64bit)
libQt5DBus.so.5(Qt_5)(64bit)
libQt5DBus.so.5(Qt_5_PRIVATE_API)(64bit)
libQt5Gui.so.5()(64bit)
libQt5Gui.so.5(Qt_5)(64bit)
libQt5Widgets.so.5()(64bit)
libQt5Widgets.so.5(Qt_5)(64bit)
libc.so.6()(64bit)
libc.so.6(GLIBC_2.14)(64bit)
libc.so.6(GLIBC_2.17)(64bit)
libc.so.6(GLIBC_2.2.5)(64bit)
libc.so.6(GLIBC_2.4)(64bit)
libgcc_s.so.1()(64bit)
libgcc_s.so.1(GCC_3.0)(64bit)
libm.so.6()(64bit)
libstdc++.so.6()(64bit)
libstdc++.so.6(CXXABI_1.3)(64bit)
libstdc++.so.6(GLIBCXX_3.4)(64bit)
polkit
qt5-qtbase
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
rtld(GNU_HASH)
systemd

**Comment by omni6** [comment]
I am the maintainer of the dnscrypt-proxy-gui [PKGBUILD](https://aur.archlinux.org/packages/dnscrypt-proxy-gui/) and [this](https://pastebin.com/raw/VcFByfhs) are all pulled dependencies of this package. I think everything should be satisfied. They will be automatic installed from arch package manager when using my PKGBUILD. Dnscrypt-proxy-gui was first built in clean chroot before uploaded to AUR. i think you can exclude the case of a missing lib because most arch users will use that pkgbuild. 

Can noone provide a proper **backtrace**??

**Comment by F1ash** [comment]
I'll try in the coming days ...

**Comment by F1ash** [comment]
For beginning.
Try build and run in terminal from `debug` branch. I'm wait for results.

**Comment by F1ash** [comment]
i'm get that backtrace in terminal at start:

readSettings
firstServiceStart
changeUnitsFinished
startService
INACTIVE/FAILED
STOP_SLICE
......
INACTIVE/FAILED
ACTIVE
at close:
changeVisibility 0
INACTIVE/FAILED
closeEvent
RESTORED

 What you have for comparison?

**Comment by omni6** [comment]
My output to compare:

```
~ » DNSCryptClient                                                                                                                                                                                       jan@royal-ts
readSettings
firstServiceStart
changeUnitsFinished
changeVisibility 0
startService
INACTIVE/FAILED
ACTIVE
changeVisibility 1
changeVisibility 0
STOP_SLICE
STOP_SLICE
INACTIVE/FAILED
```
as said, i have no crash!

**Comment by omni6** [comment]
@BooBerry @nsiregar @milkmiruku @unixfox

**Comment by F1ash** [comment]
updated `debug` branch for check tray icon stability;

**Comment by mxmilkiib** [comment]
debug branch after a `sudo make install` and `rm ~/.config/DNSCryptClient/DNSCryptClient.conf` gives:
```
qt5ct: using qt5ct plugin
qt5ct: D-Bus system tray: no
readSettings
firstServiceStart
changeUnitsFinished
```
tray icon appears for a fraction of a second then dissapears.

```unset QT_QPA_PLATFORMTHEME```
= same, without qt5ct msgs

[strace](https://gist.github.com/milkmiruku/ecdb2201d5926547b0d2cc1a59c298f3#file-dcc-log-txt) / [ltrace](https://gist.github.com/3002833d6e2cbee0771f8ba612fad79b)

the ltrace log is straight through, though whilst running the ltrace (~1m16s), the tray icon stays for much longer and I can open the program if I wish. if i keep the window open, the program doesn't seem to exit. 

if i let the ltrace go straight through, the icon stays for the most part, there are bursts of activity roughly every half second which appears to finish with changeUnitsFinished, at which point the tray icon goes and the program exits.

anything else i could try/report?

**Comment by omni6** [comment]
@milkmiruku 

```
access("/usr/local/share/icons/hicolor/48x48/stock/navigation/DNSCryptClient_start.svg", F_OK) = -1 ENOENT (No such file or directory)
access("/usr/local/share/icons/hicolor/48x48/stock/net/DNSCryptClient_start.png", F_OK) = -1 ENOENT (No such file or directory)
access("/usr/local/share/icons/hicolor/48x48/stock/net/DNSCryptClient_start.svg", F_OK) = -1 ENOENT (No such file or directory)
access("/usr/local/share/icons/hicolor/48x48/stock/object/DNSCryptClient_start.png", F_OK) = -1 ENOENT (No such file or directory)
access("/usr/local/share/icons/hicolor/48x48/stock/object/DNSCryptClient_start.svg", F_OK) = -1 ENOENT (No such file or direc
```
i think you can avoid ^this^ if you use:

`cmake -DCMAKE_INSTALL_PREFIX=/usr -DSHARE_INSTALL_PREFIX=/usr/share ..`

**Comment by mxmilkiib** [comment]
Built it with that but no apparent change.

**Comment by omni6** [comment]
Maybe for all that are not skilled to compile from source, here a [git-debug](https://pastebin.com/raw/aJPNLkjG) PKGBUILD. 
It will build a package from the latest debug branch and install it as dnscrypt-proxy-gui-debug. This is all i can do ;)
   use it with `makepkg -si`

**Comment by mxmilkiib** [comment]
Built the pkg and installed, no difference.

**Comment by F1ash** [comment]
updated `debug` branch: probe to show app forcibly;
still wait for results...

**Comment by F1ash** [comment]
guys, i can't reproduce your problem, therefore i wait your results with new updates...

**Comment by mxmilkiib** [comment]
What ever has changed in debug, it starts (or rather, stays started) for me now. Fantastic, thanks for the help :)

**Comment by F1ash** [comment]
good... it is not a fantastic. i call show() method for main window. I don't know but in the fedora of Qt5-framework usually not need to call show() for main window... then i'll add this changes to `master` branch (some later).

**Comment by unixfox** [comment]
It's working for me too.
I'm using Plasma 5 on ArchLinux.
